src/Controller/GoogleAuthController.php

```diff
@@ -20,50 +22,75 @@ class GoogleAuthController extends ControllerBase {
    */
-  protected $networkManager;
+  private $networkManager;
+
+  /**
+   * The user manager.
+   *
+   * @var \Drupal\social_auth\SocialAuthUserManager
+   */
+  private $userManager;
 
   /**
-   * The Google authentication manager.
+   * The google authentication manager.
    *
    * @var \Drupal\social_auth_google\GoogleAuthManager
    */
-  protected $googleManager;
+  private $googleManager;
 
   /**
-   * The user manager.
+   * Used to access GET parameters.
    *
-   * @var \Drupal\social_auth\SocialAuthUserManager
+   * @var \Symfony\Component\HttpFoundation\RequestStack
    */
-  protected $userManager;
+  private $request;
 
   /**
-   * The session manager.
+   * The Social Auth Data Handler.
    *
-   * @var \Symfony\Component\HttpFoundation\Session\SessionInterface
+   * @var \Drupal\social_auth\SocialAuthDataHandler
    */
-  protected $session;
+  private $dataHandler;
+
+
+  /**
+   * The logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $loggerFactory;
 
   /**
-   * GoogleLoginController constructor.
+   * GoogleAuthController constructor.
    *
    * @param \Drupal\social_api\Plugin\NetworkManager $network_manager
    *   Used to get an instance of social_auth_google network plugin.
    * @param \Drupal\social_auth\SocialAuthUserManager $user_manager
    *   Manages user login/registration.
    * @param \Drupal\social_auth_google\GoogleAuthManager $google_manager
    *   Used to manage authentication methods.
-   * @param SessionInterface $session
-   *   Used to store the access token into a session variable.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request
+   *   Used to access GET parameters.
+   * @param \Drupal\social_auth\SocialAuthDataHandler $social_auth_data_handler
+   *   SocialAuthDataHandler object.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   Used for logging errors.
    */
-  public function __construct(NetworkManager $network_manager, SocialAuthUserManager $user_manager, GoogleAuthManager $google_manager, SessionInterface $session) {
+  public function __construct(NetworkManager $network_manager, SocialAuthUserManager $user_manager, GoogleAuthManager $google_manager, RequestStack $request, SocialAuthDataHandler $social_auth_data_handler, LoggerChannelFactoryInterface $logger_factory) {
+
     $this->networkManager = $network_manager;
     $this->userManager = $user_manager;
     $this->googleManager = $google_manager;
-    $this->session = $session;
+    $this->request = $request;
+    $this->dataHandler = $social_auth_data_handler;
+    $this->loggerFactory = $logger_factory;
+
     // Sets the plugin id.
     $this->userManager->setPluginId('social_auth_google');
+
     // Sets the session keys to nullify if user could not logged in.
-    $this->userManager->setSessionKeysToNullify(['social_auth_google_access_token']);
+    $this->userManager->setSessionKeysToNullify(['access_token']);
```

Comment:
Change it by $this->userManager->setSessionKeysToNullify(['access_token', 'oauth2state']);